From 24b5363b450410fc4b8332a3089d1e3810f4f7a8 Mon Sep 17 00:00:00 2001
From: Mingcong Bai <jeffbai@aosc.io>
Date: Thu, 4 Sep 2025 15:33:15 +0800
Subject: [PATCH 78/90] ia64/mm: Make vmemmap_free() available only with
 CONFIG_MEMORY_HOTPLUG

vmemmap_free() callsites (mm/sparse.c) and declaration (include/linux/mm.h)
are protected with CONFIG_MEMORY_HOTPLUG. This function is not required if
CONFIG_MEMORY_HOTPLUG is not enabled. Hence move the config wrapper outside
the function definition.

Signed-off-by: Mingcong Bai <jeffbai@aosc.io>
Signed-off-by: Xinhui Yang <cyan@cyano.uk>
---
 arch/ia64/mm/discontig.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/arch/ia64/mm/discontig.c b/arch/ia64/mm/discontig.c
index c55f6e9b2c3a..9860b250c845 100644
--- a/arch/ia64/mm/discontig.c
+++ b/arch/ia64/mm/discontig.c
@@ -631,9 +631,10 @@ int __meminit vmemmap_populate(unsigned long start, unsigned long end, int node,
 {
 	return vmemmap_populate_basepages(start, end, node, NULL);
 }
-
+#ifdef CONFIG_MEMORY_HOTPLUG
 void vmemmap_free(unsigned long start, unsigned long end,
 		struct vmem_altmap *altmap)
 {
 }
 #endif
+#endif /* CONFIG_SPARSEMEM_VMEMMAP */
-- 
2.25.1

